WordPress/Classes/Utility/BackgroundTasks/WeeklyRoundupBackgroundTask.swift

```diff
@@ -427,25 +511,15 @@ class WeeklyRoundupBackgroundTask: BackgroundTask {
                 return
             }
 
-            for (site, stats) in siteStats {
+            for (siteID, stats) in siteStats {
                 group.enter()
-
-                self.notificationScheduler.scheduleDynamicNotification(
-                    site: site,
-                    views: stats.viewsCount,
-                    comments: stats.commentsCount,
-                    likes: stats.likesCount,
-                    periodEndDate: self.currentRunPeriodEndDate()
-                ) { result in
-
-                    switch result {
-                    case .success:
-                        self.eventTracker.notificationScheduled(type: .weeklyRoundup, siteId: site.dotComID?.intValue)
-                    case .failure(let error):
-                        onError(error)
+                self.coreDataStack.performAndSave { context in
+                    self.scheduleDynamicNotification(siteID: siteID, stats: stats, in: context) { result in
```

Comment:
We can remove the context argument from the `scheduleDynamicNotification` function, and let the function itself uses `coreDataStack.performQuery` to find the `Blog` instance using the `siteID` argument.

Reply:
@crazytonyli Same question as the one from this [comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/21076#discussion_r1268800180). Is there a particular reason why fetching the blog from the main thread is a better approach? Even though this is a "background task".

Reply:
I have answered the question of `performAndSave` or `performQuery` in https://github.com/wordpress-mobile/WordPress-iOS/pull/21076#discussion_r1268829526. What do you think removing the context argument from `scheduleDynamicNotification`?

Reply:
> What do you think removing the context argument from scheduleDynamicNotification?

Sounds good, I'll make the change tomorrow.

Reply:
Resolved in f2858c9088c6109b4ec266208c650891298e7a36